### ISSUE TITLE
GGM - BUG - Objects in front of other objects you have multi-selected get the same glow/outline as if they are also selected

### ISSUE THREAD

**Issue by Kravenwolf** [TheGameCreators/GameGuruRepo#4283]
This can get confusing, as it leads to you thinking you may have accidentally selected them when you did not. Notice the three objects in front of the walls I multi-select also glow orange as if they were selected, too as long as those selected objects remain behind them.
 
https://user-images.githubusercontent.com/103399301/227107418-35095362-1f44-422b-b7d2-1208f84cf95d.mp4

**Comment by MonkeyFrogStudio** [comment]
Yep. It can be confusing at times. It would be nicer if this worked similarly to how it works in Blender, I think.

Nothing selected:
![select1](https://user-images.githubusercontent.com/36735507/227219354-af43b0c0-47c6-4f22-a1fd-8857a3306296.png)

Front object selected:
![select2](https://user-images.githubusercontent.com/36735507/227219395-ecc020e1-6508-4f2e-bd9f-6cd41871e4ae.png)

Back object selected:
![select3](https://user-images.githubusercontent.com/36735507/227219446-10c09e32-cb62-4236-9da2-eae321b36c94.png)

Then there's never any confusion as to what's selected.

**Comment by LeeBamberTGC** [comment]
@Kravenwolf This is a piggy back off the current Wicked Engine implementation for highlighting things. No easy way to change it without a rewrite as far as I can see. Setting this as a feature request that can replace the current system with one more like Blender. The current system sees the foreground objects in the zbuffer and uses that to make the stencil:
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/35840583/9b1bc358-c776-42a2-91cc-20a05d6bcd46)

It kind of works when objects are transparent and double sided, but that is only because they have no presence in the stencil buffer:
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/35840583/64f2f1c1-f24a-4b4e-be28-86146a90ddc9)

The new Wicked Engine may solve this, or there might be a fudge we can try with the current WE once bugs are out of the way.
